server/cli/src/refresh_dates.rs

```diff
@@ -277,14 +288,17 @@ impl<'a> RefreshDatesRepository<'a> {
         &self,
         table_name: &str,
         field_name: &str,
+        store_ids: Option<String>,
     ) -> Result<Vec<IdAndTimestamp>, RepositoryError> {
         // the program_event table is using `9999-09-09 09:09:09` as a max timestamp value
         // we don't want to update this datetime value
-        let query = format!(
-            "select id, {} as dt from {} where {0} is not null and {0} <> '9999-09-09 09:09:09'",
-            field_name, table_name
-        );
-
+        let query = if let Some(store_ids) = store_ids {
+            format!(
+            "select id, {field_name} as dt from {table_name} t join changelog cl on cl.record_id = t.id where cl.store_id in {store_ids} and {field_name} is not null and {field_name} <> '9999-09-09 09:09:09'")
```

Comment:
```suggestion
            "select id, {field_name} as dt from {table_name} t join changelog cl on cl.record_id = t.id where t.table_name = {table_name} and cl.store_id in {store_ids} and {field_name} is not null and {field_name} <> '9999-09-09 09:09:09'")
```
I think there's a small chance of record id being repeated. For example masterlist/program? or manually entered data, `name_oms_fields` but that being said I think it's probably fine now...